src/assets/js/dynamic-addon-cards.js

```diff
@@ -33,6 +35,26 @@
       button.addEventListener('click', (event) => {
         event.preventDefault();
       });
+    } else if (
+      typeof navigator !== 'undefined' &&
+      typeof navigator.mozAddonManager !== 'undefined'
+    ) {
+      button.addEventListener('click', async (event) => {
+        event.preventDefault();
+
+        try {
+          const installObj = await navigator.mozAddonManager.createInstall({
+            url: downloadURL,
+            hash: fileHash,
+          });
+
+          await installObj.install();
+        } catch (e) {
+          console.debug(
+            `failed to install add-on with addonId=${addonId}: ${e.message}`
```

Comment:
Should we show anything to the user at this point?

Reply:
This happens when there is a signing issue or a problem during the download. I think we might eventually but we don't have any notices on the add-on cards yet. That's something I wanted to discuss with UX and Product (as a mvp-polish thing).

Reply:
I filed an issue for that.